src/librustc_traits/lib.rs

```diff
@@ -51,13 +48,13 @@ pub fn provide(p: &mut Providers) {
         program_clauses_for: lowering::program_clauses_for,
         program_clauses_for_env: lowering::program_clauses_for_env,
         evaluate_obligation: evaluate_obligation::evaluate_obligation,
-        type_op_eq: type_op_eq::type_op_eq,
-        type_op_prove_predicate: type_op_prove_predicate::type_op_prove_predicate,
-        type_op_subtype: type_op_subtype::type_op_subtype,
-        type_op_normalize_ty: type_op_normalize::type_op_normalize_ty,
-        type_op_normalize_predicate: type_op_normalize::type_op_normalize_predicate,
-        type_op_normalize_fn_sig: type_op_normalize::type_op_normalize_fn_sig,
-        type_op_normalize_poly_fn_sig: type_op_normalize::type_op_normalize_poly_fn_sig,
+        type_op_eq: type_op::type_op_eq,
+        type_op_prove_predicate: type_op::type_op_prove_predicate,
+        type_op_subtype: type_op::type_op_subtype,
+        type_op_normalize_ty: type_op::type_op_normalize_ty,
+        type_op_normalize_predicate: type_op::type_op_normalize_predicate,
+        type_op_normalize_fn_sig: type_op::type_op_normalize_fn_sig,
+        type_op_normalize_poly_fn_sig: type_op::type_op_normalize_poly_fn_sig,
```

Comment:
I'd suggest having more `provide` methods, one per module, even if only for using `foo,` instead of `foo: foo::foo,`.

Reply:
Done